app/Support/PluginManager.php

```
@@ -112,10 +112,22 @@ public function autoRegisterBundledPlugins(): int
-                            $this->runPluginMethod($pluginName, 'onActivate');
+                            $upgradeInstance = $this->instantiatePlugin([
+                                'id'        => (int) $row['id'],
+                                'name'      => $pluginName,
+                                'path'      => $pluginName,
+                                'main_file' => $pluginMeta['main_file'] ?? 'wrapper.php',
+                            ]);
+                            if (method_exists($upgradeInstance, 'onActivate')) {
+                                $upgradeInstance->onActivate();
+                            }
                         } catch (\Throwable $e) {
                             SecureLogger::warning("[PluginManager] Note: onActivate failed during upgrade for $pluginName: " . $e->getMessage());
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'App\Support\SecureLogger' in method 'autoRegisterBundledPlugins'. (undefined)

(StaticAccess)